src/ResourceManager/Sql/Commands.Sql/VirtualNetworkRule/Services/AzureSqlServerVirtualNetworkRuleAdapter.cs

```diff
@@ -0,0 +1,130 @@
+        /// <param name="serverName">The name of the server</param>
+        /// <returns>A list of all the virtual network rules</returns>
+        public List<AzureSqlServerVirtualNetworkRuleModel> ListVirtualNetworkRules(string resourceGroupName, string serverName)
+        {
+            var resp = Communicator.List(resourceGroupName, serverName);
+            return resp.Select((s) =>
+            {
+                return CreateVirtualNetworkRuleModelFromResponse(resourceGroupName, serverName, s);
+            }).ToList();
+        }
+
+        /// <summary>
+        /// Upserts a virtual network rule
+        /// </summary>
+        /// <param name="resourceGroup">The name of the resource group</param>
+        /// <param name="serverName">The name of ther server</param>
+        /// <param name="model">The virtual network rule to create</param>
+        /// <returns>The updated server model</returns>
+        public AzureSqlServerVirtualNetworkRuleModel UpsertVirtualNetworkRule(AzureSqlServerVirtualNetworkRuleModel model)
+        {
+            var resp = Communicator.CreateOrUpdate(model.ResourceGroupName, model.ServerName, model.VirtualNetworkRuleName, new Management.Sql.Models.VirtualNetworkRule()
+            {
+                VirtualNetworkSubnetId = model.VirtualNetworkSubnetId
+            });
+            return CreateVirtualNetworkRuleModelFromResponse(model.ResourceGroupName, model.ServerName, resp);
+        }
+
+        /// <summary>
+        /// Deletes a virtual network rule
+        /// </summary>
+        /// <param name="resourceGroupName">The resource group the server is in</param>
+        /// <param name="serverName">The name of the server from which the virtual network rule should be removed</param>
+        /// <param name="vnetFirewallRuleName">The name of the virtual network rule to remove</param>
+        public void RemoveVirtualNetworkRule(string resourceGroupName, string serverName, string vnetFirewallRuleName)
+        {
+            Communicator.Remove(resourceGroupName, serverName, vnetFirewallRuleName);
+        }
+
+        /// <summary>
+        /// Convert a Management.Sql.Models.VnetFirewallRule to AzureSqlServerVirtualNetworkRuleModel
+        /// </summary>
+        /// <param name="resourceGroup">The resource group the server is in</param>
+        /// <param name="serverName">The name of the server</param>
+        /// <param name="resp">The management client server response to convert</param>
+        /// <returns>The converted server model</returns>
+        private static AzureSqlServerVirtualNetworkRuleModel CreateVirtualNetworkRuleModelFromResponse(string resourceGroup, string serverName, Management.Sql.Models.VirtualNetworkRule resp)
+        {
+            AzureSqlServerVirtualNetworkRuleModel vnetFirewallRuleName = new AzureSqlServerVirtualNetworkRuleModel();
+
+            vnetFirewallRuleName.ResourceGroupName = resourceGroup;
```

Comment:
minor style comment, low priority: I like to use [object initializer syntax](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/object-and-collection-initializers) in cases like this